firestore/FirestoreSwiftUIExample/Views/ContentView.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  ContentView.swift
+//  FirestoreSwiftUIExample
+//
+
+import SwiftUI
+import Firebase
+
+struct ContentView: View {
+    let db = Firestore.firestore()
+    @ObservedObject var restaurantListViewModel = RestaurantListViewModel()
+
+    var body: some View {
+        NavigationView {
+            List(restaurantListViewModel.restaurants) { restaurant in
+                NavigationLink(destination: RestaurantDetailView()) {
+                    RestaurantItemView(restaurant: restaurant)
+                }
+            }
+            .navigationBarTitle("Friendly Eats", displayMode: .inline)
+            .onAppear() {
+                self.restaurantListViewModel.fetchData()
+            }
+            .toolbar {
+                ToolbarItem(placement: .navigationBarLeading) {
+                    Button("Populate") {
+                        populate()
+                    }
+                }
+
+                ToolbarItem(placement: .navigationBarTrailing) {
+                    Button("Filter") {
+                        print(restaurantListViewModel.restaurants.count)
+                    }
+                }
+            }
+        }
+    }
+
+    func populate() {
```

Comment:
Similar to Peter's comment above, this method should (if we're being clean with our architecture) live outside of the view layer. This wasn't the case when we wrote the original app, so you don't have to clean it up here, but a good place to put it would be on an extension of the Firestore class.

```swift
extension Firestore {

  func populate() {
    // ...
  }

}
```